src/component/child/index.js

```diff
@@ -76,6 +80,23 @@ export class ChildComponent extends BaseComponent {
         });
     }
 
+    hasValidParentDomain() {
+        const parentDomain = this.getParentDomain();
+        if (typeof this.component.allowedParentDomains === 'string') {
```

Comment:
Given results of these tests, ```matchDomain ``` needs some attention:
``` javascript
import { matchDomain } from 'cross-domain-utils/src';

describe('match domain', () => {
    describe('returns true when', () => {
        it('domain is array of regex and origin is array of regex', () => {
            assert.isTrue(matchDomain([/^http\:\/\/domain$/], [/^http\:\/\/domain$/]));
        });

        it('domain is a string and origin is a string', () => {
            assert.isTrue(matchDomain(['http://domain'], 'http://domain'));
        });

        it('domain is array of regex and origin is a string', () => {
            assert.isTrue(matchDomain([/^http\:\/\/domain$/], 'http://domain'));
        });

        it('domain is regex and origin is a string', () => {
            assert.isTrue(matchDomain(/^http\:\/\/domain$/, 'http://domain'));
        });
        
        it('domain is array with a wildcard and origin is a string', () => {
            assert.isTrue(matchDomain(['*'], 'http://domain'));
        });

        it('domain is array of strings and origin is a string', () => {
            assert.isTrue(matchDomain(['http://domain'], 'http://domain'));
        });
        
    });
});
```
match domain
    returns true when
      √domain is array of regex and origin is array of regex
      √domain is a string and origin is a string
      ×domain is array of regex and origin is a string
        expected false to be true
      ×domain is regex and origin is a string
        expected [ Array(1) ] to be true
      ×domain is array with a wildcard and origin is a string
        expected false to be true
      √domain is array of strings and origin is a string
```

```